### ISSUE TITLE
Restart & Run All

### ISSUE THREAD

**Issue by mikaem** [executablebooks/thebe#237]
Hi

First of all, thanks a lot for making a great piece of software:-) I am currently using jupyter book with thebe for both my [teaching](https://mikaem.github.io/MEK4300/content/intro.html) and for documentation of my [software](https://mikaem.github.io/shenfun-demos/content/intro.html).

I just have a small feature request. Would it be possible to add a Restart & Run All button for thebe/jupyter book? I use this feature in most of my notebooks all the time, and I think it would make my documentation much better and easier to use as well. 

Thanks a lot:-)

**Comment by choldgraf** [comment]
Definitely - I think that this could be an addition somewhere around here:

https://github.com/executablebooks/thebe/blob/master/src/thebelab.js#L189

we'd need to figure out the right JupyterLab API call to trigger this as well.

**Comment by manics** [comment]
It looks like it's not a single command, instead it's a dialog that restarts the session before running all cells:
https://github.com/jupyterlab/jupyterlab/blob/09b739d44d5dee6ad44b0243871c12b772b1b2d8/packages/notebook-extension/src/index.ts#L2299-L2309

**Comment by rowanc1** [comment]
I am starting to look into this in ~30 min!

**Comment by choldgraf** [comment]
Two quick thoughts on buttons etc:

At some point we may want to rethink the UIUX of buttons for thebe cells so they are not overloaded with buttons. Eg either make buttons configurable or use a drop-down for some?

**Comment by rowanc1** [comment]
Right now I am going with something like this:

![image](https://user-images.githubusercontent.com/913249/96351533-2901c080-1079-11eb-86a4-2ece68431c99.png)

Happy to make changes once I get a bit more oriented in what I am actually doing ...... 🧐

**Comment by choldgraf** [comment]
I think for now it makes sense to just go with that, UI/UX is probably another enhancement for the future and we might wanna think about it a little bit. We want to find the right balance between "things many people will want", and "fully recreating the notebook UI" lol

**Comment by rowanc1** [comment]
Yep, this is the most minimally invasive approach for now. And mostly just me getting oriented in the lib.

**Comment by moorepants** [comment]
Both "restart" and "restart and run all" should probably be displayed only once on a page. Would it be possible to have those two buttons only on the first cell? Maybe even above the first cell? Or should it be like the "activate" button, in that you only display it if you want the user to have that functionality.

**Comment by rowanc1** [comment]
Can we tackle this outside of this small pr?

I think it would be good to have some small status indicators as well? Is the kernal busy, etc. But that is a bit more work and requires maybe a restructure of some of the functions that are scoped inside of `renderCell`.

**Comment by choldgraf** [comment]
I'm +1 on tackling this in a different PR. We'll wanna change the UI around this eventually but for now I'm a fan of celebrating the baby steps :-)  

Maybe we can open a new PR to discuss UI/UX improvements in general?

**Comment by rowanc1** [comment]
Have put a PR together, that I think works, but would appreciate a sanity check from someone with more experience in the lib!
